libsnark/gadgetlib1/protoboard.tcc

```diff
@@ -116,6 +116,8 @@ void protoboard<FieldT>::augment_variable_annotation(const pb_variable<FieldT> &
     auto it = constraint_system.variable_annotations.find(v.index);
     constraint_system.variable_annotations[v.index] = (it == constraint_system.variable_annotations.end() ? "" : it->second + " ") + postfix;
 #endif
+    (void)v;
```

Comment:
Move into a `#else` to have something like
```
#ifdef DEBUG
    <if block>;
#else
    libff::UNUSED(v);
    libff::UNUSED(postfix);
#endif
```

Reply:
Let's follow a similar approach for the other places where you "void casted" in order to be consistent with unused variable handling.

Reply:
OK.  I've tried to add the `#ifdef` blocks in.  Sometimes, they use `DEBUG` in the, in which case I do exactly as in your comment.  In some cases, we have to use `NDEBUG` in the conditional, where a variable is only defined to be used in an assert.